test/SILOptimizer/cast_folding_conditional_conformance.swift

```diff
@@ -0,0 +1,18 @@
+// RUN: %target-swift-frontend %s -emit-sil -O -o - | %FileCheck %s
+// RUN: %target-swift-frontend %s -emit-ir -o /dev/null
+
+// rdar://problem/38694450
+
+protocol P {}
+extension Array: P where Element: P {}
+struct X {}
+// CHECK: unconditional_checked_cast_addr Array<X> in {{%[0-9]*}} : $*Array<X> to P in {{%[0-9]*}} : $*P
+let x = [X()] as! P
+
+struct Y<T> {}
+extension Y: P where T: P {}
+public func foo<T>(_: T.Type) {
+    // CHECK: unconditional_checked_cast_addr Y<T> in {{%[0-9]*}} : $*Y<T> to P in {{%[0-9]*}} : $*P // id: %7
```

Comment:
Don't match the `// id` its too specific

Reply:
I know, I just missed it in this case. Thanks for pointing it out.